packages/ui-kit/src/helpers/graphql/stories/DataPoolInput.stories.tsx

```diff
@@ -0,0 +1,24 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import type { DataPoolInput } from '../index'
+
+export const DataPoolInputComponent: React.FC<DataPoolInput> = () => null
+
+const meta = {
```

Comment:
In GraphQL, we use `@oneOf` for this type. In TypeScript, it would be best defined as

```ts
type DataPoolInput
  = { id: string }
  | { name: string }
```

Should we use this type instead? Is it ergonomic? Does Storybook support it?

Reply:
the thing is that about the `@oneOf` directive,  when we use it, it generates the type like below,

```
 type DataPoolInput 
 = { id: string; name?: never; }
  | { id?: never; name: string; }; 
```
It does not only gives one option. provides both id and name. So, we need to apply a solution for `@oneOf` generation.

Reply:
What if we write a custom, equivalent type, though? I.e., don't use the `@oneOf` generated type. Use our own instead?

Reply:
When I tried a custom type here without generating `@oneOf`, But still it gives the same result.
So, I added the note storybook